app.py

```diff
@@ -939,38 +982,81 @@ def snap_release(snap_name):
 @app.route('/account/snaps/<snap_name>/market/', methods=['POST'])
 @login_required
 def post_market_snap(snap_name):
-    whitelist = [
-        'title',
-        'summary',
-        'description',
-        'contact_url',
-        'keywords',
-        'license',
-        'price',
-        'blacklist_countries',
-        'whitelist_countries'
-    ]
-
     if 'submit_revert' in flask.request.form:
         flask.flash("All changes reverted.", 'information')
     else:
+        error_list = []
+        info = []
+        files = []
+        images_json = None
+
+        icon = flask.request.files.get('icon')
+        if icon is not None:
+            info.append(build_image_info(icon, 'icon'))
+            files.append(icon)
+
+        screenshots = flask.request.files.getlist('screenshots')
+        for screenshot in screenshots:
+            info.append(build_image_info(screenshot, 'screenshot'))
+            files.append(screenshot)
+
+        if not files:
```

Comment:
Maybe it would be better to consistently use the word "screenshot" in related variable names? 'cos we're probably going to be dealing with a few different types of data in this method. E.g. `screenshot_files`, `screenshots_json`. Or, maybe it would make sense to encapsulate all the screenshot work into its own function (`_get_screenshots` or something) - in that case the shorter contextual variable names would be fine.